src/instructlab/sdg/pipeline.py

```diff
@@ -32,6 +33,32 @@ def __init__(
         self.num_procs = 8
 
 
+# This is part of the public API.
+class BlockGenerationError(Exception):
```

Comment:
I suggested this, so sorry I'm changing - `PipelineBlockError` if it wraps the block instantiation too?

Also, add it to `__all__` in `__init__.py`